packages/frontend-2/components/settings/workspaces/General/DeleteDialog.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+  <LayoutDialog
+    v-model:open="isOpen"
+    title="Delete workspace"
+    max-width="sm"
+    :buttons="dialogButtons"
+  >
+    <p>
+      Are you sure you want to
+      <span class="font-medium">permanently delete</span>
+      the selected workspace?
+    </p>
+    <div
+      class="rounded border bg-foundation-2 border-outline-3 text-body-2xs text-foreground font-medium py-3 px-4 my-4"
+    >
+      {{ workspace.name }}
+    </div>
+    <p>
+      This action
+      <span class="font-medium">cannot</span>
+      be undone.
+    </p>
+  </LayoutDialog>
+</template>
+<script setup lang="ts">
+import { useMutation } from '@vue/apollo-composable'
+import { deleteWorkspaceMutation } from '~/lib/settings/graphql/mutations'
+import type { LayoutDialogButton } from '@speckle/ui-components'
+import { graphql } from '~~/lib/common/generated/gql'
+import {
+  convertThrowIntoFetchResult,
+  getFirstErrorMessage
+} from '~~/lib/common/helpers/graphql'
+import { homeRoute } from '~/lib/common/helpers/route'
+import type { SettingsWorkspaceGeneralDelete_WorkspaceFragment } from '~~/lib/common/generated/gql/graphql'
+import { ToastNotificationType, useGlobalToast } from '~~/lib/common/composables/toast'
+
+graphql(`
+  fragment SettingsWorkspaceGeneralDelete_Workspace on Workspace {
+    id
+    name
+  }
+`)
+
+const props = defineProps<{
+  workspace: SettingsWorkspaceGeneralDelete_WorkspaceFragment
+}>()
+
+const isOpen = defineModel<boolean>('open', { required: true })
+
+const { mutate: deleteWorkspace } = useMutation(deleteWorkspaceMutation)
+const { triggerNotification } = useGlobalToast()
+
+const onDelete = async () => {
+  const result = await deleteWorkspace({
+    workspaceId: props.workspace.id
+  }).catch(convertThrowIntoFetchResult)
+
+  if (result?.data) {
+    isOpen.value = false
+    triggerNotification({
+      type: ToastNotificationType.Success,
+      title: 'Workspace deleted',
+      description: `The ${props.workspace.name} workspace has been deleted`
+    })
+    navigateTo(homeRoute)
```

Comment:
The workspace is still in the local apollo cache tho, so it may appear elsewhere in the site (e.g. maybe settings workspace list).

You need to do a cache mutation where you evict the Workspace object from cache, and possibly even alter the totalCount of known workspace lists (e.g. User.workspaces)

```
// Evict object from cache
cache.evict({
  id: getCacheId('Workspace', workspaceId)
})

modifyObjectFields(cache, getCacheId('User', activeUserId), () => {..}, {fieldwhitelist: ['workspaces]})
```

Reply:
Good point, thanks for for snippet. I completely forgot about this, I've updated it to remove the item from the cache and update the total count